app/controllers/DucrPartChiefController.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions.{AuthAction, DucrPartsAction, JourneyRefiner, NonIleQueryAction}
+import forms.DucrPartChiefChoice
+import forms.DucrPartChiefChoice.form
+import javax.inject.{Inject, Singleton}
+import models.ReturnToStartException
+import models.cache.JourneyType.JourneyType
+import models.cache._
+import models.requests.JourneyRequest
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import repositories.CacheRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.ducr_part_chief
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class DucrPartChiefController @Inject()(
+  authenticate: AuthAction,
+  getJourney: JourneyRefiner,
+  isDucrPartsFeatureEnabled: DucrPartsAction,
+  ileQueryFeatureDisabled: NonIleQueryAction,
+  cacheRepository: CacheRepository,
+  mcc: MessagesControllerComponents,
+  ducrPartChiefPage: ducr_part_chief
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
+
+  def displayPage(): Action[AnyContent] =
+    (authenticate andThen isDucrPartsFeatureEnabled andThen ileQueryFeatureDisabled andThen getJourney(
+      JourneyType.ARRIVE,
+      JourneyType.DEPART,
+      JourneyType.ASSOCIATE_UCR,
+      JourneyType.DISSOCIATE_UCR
+    )) { implicit request =>
```

Comment:
Just a suggestion. This can be extracted to the variable. You're using it in two places and it's quite big, few lines, but not sure about the name for it.

```
val requiredActions = (authenticate andThen isDucrPartsFeatureEnabled andThen ileQueryFeatureDisabled andThen getJourney(
      JourneyType.ARRIVE,
      JourneyType.DEPART,
      JourneyType.ASSOCIATE_UCR,
      JourneyType.DISSOCIATE_UCR
    ))
```

and you will be able to use it as `requiredActions { implicit request => .... }` or `requiredActions.async { implicit request => ... }`

Reply:
Good idea